File: 剑指offer（c++）/数组中出现次数超过一半的数字.cpp
class Solution {
public:
    int MoreThanHalfNum_Solution(vector<int> numbers) {
        int num = numbers.size();
        int count[num+1];
        int i;
        for(i=1;i<=num;i++)
            count[i]=0;
        for(i=0;i<num;i++)
        {
                count[numbers[i]]++;
                if(count[numbers[i]] > num/2) return numbers[i];
        }
        return 0;
    }
};

File: 剑指offer（python）/数组中出现次数超过一半的数字.py
# -*- coding:utf-8 -*-
from collections import Counter 
class Solution:
    def MoreThanHalfNum_Solution(self, numbers):
        # write code here
        return Counter(numbers).most_common(1)[0][0] if Counter(numbers).most_common(1)[0][1]*2 > len(numbers) else 0